src/toil/test/src/jobCacheTest.py

```diff
@@ -0,0 +1,848 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Author : Arjun Arkal Rao
+Affiliation : UCSC BME, UCSC Genomics Institute
+File : src/toil/test/src/JobCacheTest.py
+"""
+from __future__ import print_function
+import collections
+import os
+import sys
+import random
+import signal
+import time
+import unittest
+
+from struct import pack, unpack
+from uuid import uuid4
+
+from toil.job import Job, CacheError
+from toil.test import ToilTest, needs_aws, needs_azure
+from toil.leader import FailedJobsException
+from toil.jobStores.abstractJobStore import NoSuchFileException
+
+# Some tests take too long on the AWS and Azure Job stores and are unquitable for CI.  They can be
+# be run during manual tests by setting this to False.
+testingIsAutomatic = True
+
+
+class Hidden:
+    """
+    Hiding the abstract test class from the Unittest loader so it can be inherited in different test
+    suites for the different job stores.
+    """
+    class AbstractCacheTest(ToilTest):
+        """
+        Abstract tests for the the various cache functions in Job.CachedFileStore
+        """
+        def setUp(self):
+            super(Hidden.AbstractCacheTest, self).setUp()
+            testDir = self._createTempDir()
+            self.options = Job.Runner.getDefaultOptions(self._getTestJobStorePath())
+            self.options.logLevel = 'INFO'
+            self.options.workDir = testDir
+            self.options.clean = 'always'
+            self.options.logFile = os.path.join(testDir, 'logFile')
+            self.options.disableSharedCache = False
+
+        # Sanity
+        def testToilIsNotBroken(self):
+            """
+            Runs a simple DAG to test if if any features other that caching were broken.
+            """
+            A = Job.wrapJobFn(self._uselessFunc)
+            B = Job.wrapJobFn(self._uselessFunc)
+            C = Job.wrapJobFn(self._uselessFunc)
+            D = Job.wrapJobFn(self._uselessFunc)
+            A.addChild(B)
+            A.addChild(C)
+            B.addChild(D)
+            C.addChild(D)
+            Job.Runner.startToil(A, self.options)
+
+        @staticmethod
+        def _uselessFunc(job):
+            """
+            I do nothing.  Don't judge me.
+            """
+            return None
+
+        def testExtremeCacheSetup(self):
+            """
+            Try to create the cache with bad worker active and then have 10 child jobs try to run in
+            the chain.  This tests whether the cache is created properly even when the job crashes
+            randomly.
+            """
+            self.options.retryCount = 20
+            self.options.badWorker = 0.5
+            self.options.badWorkerFailInterval = 0.1
+            for test in xrange(0, 20):
+                E = Job.wrapJobFn(self._uselessFunc)
```

Comment:
I'd accept no as an answer. It's just that Toil already has non-conventional naming conventions and this is proliferating that.